### PR TITLE
Show percentage completion in options page

### PR DESCRIPTION
**Pull request by shikharsaxena98**
- **What does this PR do?**
Solves Issue #10. Added percentages while ensuring there are no breaking changes in the app.

- **Any background context you want to provide?**

- **Screenshots and/or Live Demo**

### CODE REVIEW

File: options.js [hunk 1]
```diff
@@ -2,10 +2,15 @@
   const list = document.getElementById("saved-urls");
   var urls = "";
   chrome.storage.local.get("scroll-mark", function(result) {
+    console.log(result);
```
Comment (rafaeelaudibert) [inline options.js]: You left some debug code right here

File: options.js [hunk 2]
```diff
@@ -2,10 +2,15 @@
...
     urls = result["scroll-mark"];
     for (var url in urls) {
       var div = document.createElement("div");
-      div.innerHTML = "<a href=" + url + ">" + url + "</a>";
+      let percentage = Math.round((urls[url].offset / urls[url].total) * 100);
+      if (percentage)
+        div.innerHTML =
+          "<a href=" + url + ">" + url + "</a> (" + percentage + "% completed)";
```
Comment (rafaeelaudibert) [inline options.js]: Maybe "...% scrolled" would be more relatable.

Comment (prateek3255) [reply options.js]: Just % will do I believe

File: popup.js [hunk 3]
```diff
@@ -5,6 +5,7 @@ chrome.tabs.query({ active: true, currentWindow: true }, tabs => {
   const url = tabs[0].url;
   chrome.storage.local.get("scroll-mark", data => {
     const scrollMarkData = data["scroll-mark"];
+    const scrollMarkOffset = scrollMarkData.offset;
```
Comment (prateek3255) [inline popup.js]: Where is this being used in this file?

Comment (shikharsaxena98) [reply popup.js]: Yeah, it wasn't being used. Pushed all the changes.

File: save.js [hunk 4]
```diff
@@ -3,9 +3,10 @@
   chrome.storage.local.get("scroll-mark", data => {
     const scrollMarkData = data["scroll-mark"];
     const offset = window.pageYOffset;
+    const total = document.body.clientHeight;
     const newData = scrollMarkData
-      ? { ...scrollMarkData, [url]: offset }
-      : { [url]: offset };
+      ? { ...scrollMarkData, [url]: { offset, total } }
```
Comment (prateek3255) [inline save.js]: One last thing can you also include the title of the page in this object using `document.title`.

File: options.js [hunk 5]
```diff
@@ -5,7 +5,11 @@
     urls = result["scroll-mark"];
     for (var url in urls) {
       var div = document.createElement("div");
-      div.innerHTML = "<a href=" + url + ">" + url + "</a>";
+      let percentage = Math.round((urls[url].offset / urls[url].total) * 100);
+      if (percentage)
+        div.innerHTML =
+          "<a href=" + url + ">" + url + "</a> (" + percentage + "%)";
```
Comment (prateek3255) [inline options.js]:
```suggestion
          "<a href=" + url + ">" + title + "</a> (" + percentage + "%)";
```
Then we would be able to do something like this, which would look better I believe.

**Comment by prateek3255** [comment]
@shikharsaxena98 Just make these few changes and test it properly, other than that we are GTG.

File: options.js (hunk 5 above)
Comment (prateek3255) [reply options.js]: Also, do handle the case where the title is not available, show the URL in that case.

**Comment by shikharsaxena98** [comment]
All done.

**Comment by shikharsaxena98** [comment]
All done I think, let me know if there are any more changes.

**Comment by prateek3255** [comment]
LGTM ✨  Thanks for contributing!

**Comment by prateek3255** [comment]
@all-contributors please add shikharsaxena98 for code

**Pull request merged**